Create can_continue_execution_error.mdx (#344)
comcode-org/hackmud_wiki: docs/scripting/scripts.lib/can_continue_execution_error.mdx

---

> title: scripts.lib().can_continue_execution_error()

Reviewer: ```suggestion
title: .can_continue_execution_error()
```

---

> A function that checks if there is enough time remaining for execution based on a specified number of milliseconds. If not enough time remains, the function returns an error.

Reviewer: ```suggestion
A function wrapping [.can_continue_execution()](can_continue_execution) that checks if there is enough time remaining for execution based on a specified number of milliseconds. If not enough time remains, the function returns a user-friendly error message.
```

## Syntax

```js

> #fs.scripts.lib().can_continue_execution_error(remain_threshold);

Reviewer: ```suggestion
#fs.scripts.lib().can_continue_execution_error(remain_threshold, name);
```

```

### Parameters

> # remain_threshold

Reviewer: ```suggestion
#### remain_threshold
```

A number in milliseconds.

> 

Reviewer: ```suggestion

#### name

The name of a script / function to return if execution cannot continue.
```

### Return

> If there is not enough time remaining the function returns an object. Otherwise, it will return `null`.

Reviewer: ```suggestion
If there is not enough time remaining the function returns an object:

{ok: false, msg: name + " could not continue executing because " + remain_threshold + "ms are not available for execution."}

Otherwise, it will return `null`.
```

## Example

```js
function(context, args) {
  const l = #fs.scripts.lib();

>   const timeout_error = l.can_continue_execution_error(1000);

Reviewer: ```suggestion
  const timeout_error = l.can_continue_execution_error(1000, "my_script_name");
```
